### PR TITLE
Support Haskell number 'Infinity'

### PR DESCRIPTION
**Pull request by mchaver**
From rescript-v11 branch
https://github.com/plow-technologies/all/issues/13593
replaces this PR:
https://github.com/plow-technologies/rescript-aeson/pull/35

### CODE REVIEW

**Comment by mchaver** [comment]
@mirko-plowtech the tests are slightly different one starts with a float and goes to a float, the other starts with a string and goes to a float to mimic what the value coming from Haskell.

**Comment by mirko-plowtech** [comment]
> @mirko-plowtech the tests are slightly different one starts with a float and goes to a float, the other starts with a string and goes to a float to mimic what the value coming from Haskell.

Ohh I see now! Sorry, I was to sleepy I guess..

**Review by mirko-plowtech — APPROVED** [review]
LGTM!

File: Changelog.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 ## Changes
 
+### 6.1.0
```
Comment (mirko-plowtech) [inline Changelog.md:3]: I still think is a breaking change...but anyway is just a technicality x'D

Comment (mchaver) [reply Changelog.md]: @mirko-plowtech don't worry, I'll fix it. A lot less painful here than in `all`, no jenkins to deal with :P

**Pull request merged**